src/main/java/io/rcktapp/api/service/Configurator.java

```diff
@@ -468,7 +472,7 @@ Config findConfig() throws IOException
       {
          for (int i = -1; i <= 100; i++)
          {
-            String fileName = service.getConfigPath() + "snooze" + (i < 0 ? "" : i) + "-" + service.getProfile() + ".properties";
+            String fileName = service.getConfigPath() + "snooze" + environment + (i < 0 ? "" : i) + "-" + service.getProfile() + ".properties";
```

Comment:
We don't want to modify the two existing sections for loading properties. (because, we shouldn't have to change anything on the liftck_snooze side and it shouldn't break)

We just want to add a third new sections. So the following would work...

snooze.properties 
snooze-myapp.properties
snooze-myapp-dev.properties
snooze-myapp-prod.properties

The more specific file override the less specific file.

so add something like this...

```
if (service.getProfile() != null & service.getEnvironment() != null)
{
   // no need to do this 100 times 
   
  String fileName = service.getConfigPath() + "snooze" + "-" + service.getProfile() + "-" + service.getEnvironment() + ".properties";
      InputStream is = service.getResource(fileName);
      if (is != null)
      {
         config.files.add(fileName);
         config.props.load(is);
      }
}
```

Reply:
It wont break if we dont change anything, because if there is no environment (if is null or ""), then it would take the snooze.properties and snooze-profile.properties.
With the environment we can choose to have the default properties files (as it is now) or to use different files. 
That way we can have projects that will only have the minimun properties required in each environment file (I think thats the reason because these changes)

If we do what you said, it will take all the snooze.properties configs always and I think thats what we are trying to prevent.

We can have a call to discuss this if you want

Reply:
So..  I see what you mean ... 

```
environment = environment == null || environment.equalsIgnoreCase("") ? "" : "-" + environment;
```

So.. it is either like this..
```
snooze.properties
snooze-myapp.properties
```
or like this..
```
snooze-dev.properties
snooze-dev-myapp.properties
```

but I don't think the second part is really what we want either.

I think we want it to be like this..
```
snooze-myapp.properties   // common app props
snooze-myapp-dev.properties // dev env props
snooze-myapp-prod.properties // prod env props
```

Reply:
No, lets say there are these config files in the directory
```
snooze.properties
snooze-dev.properties
snooze-prod.properties

snooze-myapp.properties
snooze-mayapp-dev.properties
snooze-mayapp-prod.properties
```

If we dont set any environment, it will take the "default" config files as it does now ( if we set the profile to dev )
```
snooze.properties
snooze-dev.properties
```

And in the case where we set the environment to be myapp, it will take the files (if we set the dev)
```
snooze-myapp.properties
snooze-mayapp-dev.properties
```

Reply:
@tcollins I changed environment to be appName